### PR TITLE
ddl: resolve the charset by the order: table->database->server

### PR DESCRIPTION
**Pull request by winkyao**
### What problem does this PR solve? 

Before this PR:

```
mysql> create table t1(a varchar(10)) character set utf8;
Query OK, 0 rows affected (0.01 sec)

mysql> show create table t1;
+-------+----------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                 |
+-------+----------------------------------------------------------------------------------------------------------------------------------------------+
| t1    | CREATE TABLE `t1` (
  `a` varchar(10) CHARSET utf8mb4 COLLATE utf8mb4_bin DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |
+-------+----------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

Obviously, the charset of the column a should be `utf8`

### What is changed and how it works?

Resolve the charset from the table first, then the database and the last is the server.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -1269,17 +1298,36 @@ func (d *ddl) handleAutoIncID(tbInfo *model.TableInfo, schemaID int64) error {
...
+LOOP:
+	for i := len(options) - 1; i >= 0; i-- {
+		op := options[i]
+		switch op.Tp {
+		// find the last one.
+		case ast.TableOptionCharset:
+			tableCharset = op.StrValue
+			break LOOP
+		}
+	}
```
Comment (crazycs520) [inline ddl/ddl_api.go]:
```go
	for i := len(options) - 1; i >= 0; i-- {
		if options[i].Tp == ast.TableOptionCharset {
			tableCharset = options[i].StrValue
			break
		}
	}
```

**Comment by winkyao** [comment]
@crazycs520 PTAL

**Comment by crazycs520** [comment]
@winkyao Please fix CI.

File: ddl/db_integration_test.go [hunk 2]
```diff
@@ -990,6 +990,36 @@ func (s *testIntegrationSuite) TestAddIndexAfterAddColumn(c *C) {
...
+	c.Assert(err, IsNil)
+	c.Assert(tbl.Cols()[0].Charset, Equals, "latin1")
+	c.Assert(tbl.Meta().Charset, Equals, "latin1")
+
+	s.tk.MustExec(`CREATE TABLE resolve_charset1 (a varchar(255) DEFAULT NULL)`)
+	is = domain.GetDomain(ctx).InfoSchema()
+	tbl, err = is.TableByName(model.NewCIStr("resolve_charset"), model.NewCIStr("resolve_charset1"))
+	c.Assert(err, IsNil)
+	c.Assert(tbl.Cols()[0].Charset, Equals, "binary")
+	c.Assert(tbl.Meta().Charset, Equals, "binary")
```
Comment (ciscoxll) [inline ddl/db_integration_test.go]: Can you print `show create table`?

Comment (winkyao) [reply ddl/db_integration_test.go]: The check is the same with `show create table`.

**Pull request merged**